asciidtype/tests/test_asciidtype.py

```diff
@@ -177,3 +177,38 @@ def test_casting_fails_with_non_ascii_characters():
             match="Can only store ASCII text in a ASCIIDType array.",
         ):
             arr.astype(ASCIIDType(5))
+
+
+@pytest.mark.parametrize(
+    ("input1", "input2", "expected"),
+    [
+        (["hello "], ["world"], ["hello world"]),
+        (["a", "b", "c"], ["aa", "bbbb", "cc"], ["aaa", "bbbbb", "ccc"]),
```

Comment:
This is missing a tst for variable length of the first string.  And the loop seems incorrect to me:  You will have to find how much NUL-padding the first string has and start copying the other string from there.

Reply:
Good catch! Should be fixed now.